src/search/search-box/search-box.tsx

```diff
@@ -0,0 +1,85 @@
+import cn from 'classnames';
+import {
+  ChangeEvent,
+  ForwardedRef,
+  forwardRef,
+  ReactElement,
+  useContext,
+  useEffect,
+  useState,
+  useCallback,
+} from 'react';
+
+import { debounce } from '../../utils/debounce';
+import { SearchContext } from '../context/search.context';
+import { useProductSearch } from '../hooks/productSearch.hook';
+import './search-box.scss';
+
+import type { SearchBoxProps } from './search-box.types';
+
+/**
+ * The component description.
+ *
+ * ### Usage
+ *
+ * ```tsx
+ * import { SearchBox } from '@bloomreach/limitless-ui-react';
+ * import type { Configuration, ProductSearchOptions } from '@bloomreach/discovery-web-sdk';
+ *
+ * // Set the account and catalog configuration
+ * const config: Configuration = {
+ *   account_id: 1234,
+ *   domain_key: 'example_com',
+ * };
+ *
+ * // Set up the search parameters
+ * const searchOptions: ProductSearchOptions = {
+ *   q: 'Generic Metal Pants',
+ *   fl: 'pid,title,description,brand,price,thumb_image',
+ *   start: 0,
+ *   rows: 10,
+ *   url: 'http://example.com',
+ *   _br_uid_2: 'someCookieId',
+ * };
+ *
+ * export default function MyCustomComponent() {
+ *  return (
+ *    <SearchContextProvider>
+ *      <SearchBox
+ *        configuration={configuration}
+ *        searchOptions={searchOptions}
+ *        debounceDelay={300}
+ *        className="test"/>
+ *    </SearchContextProvider>
+ *  )
+ *   return <SearchBox />;
+ * }
+ * ```
+ */
+export const SearchBox = forwardRef(
+  (props: SearchBoxProps, forwardedRef: ForwardedRef<HTMLInputElement> | null): ReactElement => {
+    const { children, className, configuration, searchOptions, debounceDelay, ...rest } = props;
+    const searchContext = useContext(SearchContext);
+    const [query, setQuery] = useState<string>('');
+    const { response } = useProductSearch(query, configuration, searchOptions);
+
+    useEffect(() => {
+      searchContext.setSearchResponse(response);
+    }, [searchContext, response]);
+
+    const debouncedSetQuery = debounce((event: ChangeEvent<HTMLInputElement>) => {
+      setQuery(event.target.value);
+    }, debounceDelay ?? 500);
+
+    const onChange = useCallback(debouncedSetQuery, [debouncedSetQuery]);
+
+    return (
+      <input
```

Comment:
Wanna try the `React Banana UI` 😎

Reply:
No. Its an interesting conversation to have. I think banana ui is a great library but even if we would get into the effort of making it a headless library (aka no build-in styles and extendable templates etc), which might take a while, I think the fact that its a library primarily made for _internal_ use will clash with this libraries primary goal of being for _external_ use down the road. We are looking at Radix UI, possibly other alternatives. 

For example, lets say Design decides we should change the way the components behave tomorrow, for internal use this is fine, we can just do the effort no matter what breaks, but for external use by customers that expect the least amount of changes and upgrade efforts this would be a problem.

But that is my opinion. We still need to have a proper conversation about this with Anand too, so lets discuss in Slack.

Reply:
Yes, I agree with you. Using banana UI in the shape it is right now is a bit wrong but having something base to reuse in banana ui and here might be reasonable to think.